src/solve/mod.rs

```diff
@@ -42,118 +42,6 @@ pub enum Guidance {
 }
 
 impl Solution {
-    /// There are multiple candidate solutions, which may or may not agree on
```

Comment:
Yeah, this is dead because the chalk solver doesn't (yet?) support this kind of "advice". That is, the recursive solver would sometimes give advise to prefer answers derived from where clauses where there was ambiguity -- the slg solver doesn't draw this distinction. That's ok (imo), I'd prefer to start without it and add it back as we need to for compatibility.

(Still, worth noting, cc @scalexm)

Reply:
Yes, there are still some `FIXME` anyway, e.g. https://github.com/rust-lang-nursery/chalk/blob/master/src/solve/test/mod.rs#L1213